test/libyul/yulOptimizerTests/loopInvariantCodeMotion/no_move_immovables.yul

```diff
@@ -0,0 +1,29 @@
+{
+  let a := 1
+  function f() -> x {invalid()}
+  function g() -> y {return(0, 0)}
+  for { let i := 1 } iszero(eq(i, 10)) { a := add(i, 1) } {
+  	let b := f()
+	let c := gas()
+	let d := g()
+	let e := sload(g())
+  }
+}
+// ----
+// step: loopInvariantCodeMotion
+//
+// {
+//     let a := 1
+//     function f() -> x
+//     { invalid() }
+//     function g() -> y
+//     { return(0, 0) }
+//     let i := 1
+//     for { } iszero(eq(i, 10)) { a := add(i, 1) }
+//     {
+//         let b := f()
```

Comment:
Why exactly is `invalid` not movable? The end result is the same. Or is it just not part of this optimizer?

Reply:
because the for loop condition might be false

Reply:
Ah, true ;)